src/Parser.zig

```diff
@@ -2203,7 +2227,7 @@ fn convertInitList(p: *Parser, il: InitList, init_ty: Type) Error!NodeIndex {
             .data = .{ .bin = .{ .lhs = .none, .rhs = .none } },
         };
 
-        if (init_ty.specifier == .incomplete_array) {
+        if (init_ty.is(.incomplete_array)) {
```

Comment:
`arr_init_node == init_ty` so the below assignments should happen on a duped type (at least if init_ty is not a meta type).